app/models/user.php

```diff
@@ -10,22 +55,231 @@ public function getUserDetail($id)
         } catch (Exception $e) {
             throw new Exception("Mysql Error, record not found");
         }
+        if (empty($row['avatar'])) {
+            $row['avatar'] = '/public_images/default.jpg';
+        }
         return new self($row);
     }
+
     public static function getUserName($user_id)
     {
-        $user = new User;
-        $details = $user->getUserDetail($user_id);
-        return $details->username;
+        $data = self::getUserDetail($user_id);
+        return $data->username;
     }
+
     public static function getUserId($username)
     {
+        $db = DB::conn();
+        $row = $db->row('SELECT id FROM user WHERE username = ?', array($username));
+        if (!$row) {
+            throw new RecordNotFoundException("Mysql Error, record not found");
+        }
+        return $row['id'];
+    }
+
+    public function saveAvatar($dir)
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->update('user', array('avatar' => $dir), array('id' => $this->user_id));
+            $db->commit();
+        } catch(Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public function updateUser()
+    {
+        $this->validate();
+        if ($this->hasError()) {
+            throw new ValidationException('invalid Input');
+        }
         try {
             $db = DB::conn();
-            $row = $db->row('SELECT id FROM user WHERE username = ?', array($username));
+            $db->begin();
+            $params = array(
+                'username'   => $this->new_username,
+                'first_name' => $this->new_first_name,
+                'last_name'  => $this->new_last_name,
+                'email'      => $this->new_email,
+                );
+            $db->update('user', $params, array('id' => $this->user_id));
+            $db->commit();
+
+        } catch(Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public function changePassword()
+    {
+        try {
+        $db = DB::conn();
+        $db->begin();
+        $db->update('user', array('password' => md5($this->new_password)), array('id' => $this->user_id));
+        $db->commit();
         } catch (Exception $e) {
-            throw new Exception("Mysql Error, record not found");
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public function deleteComment($id)
+    {
+        $is_authenticated = false;
+        $db = DB::conn();
+        $db->begin();
+        $row = $db->row("SELECT * FROM comment WHERE id = ? ", array($id));
+        if ($this->user_id === $row['user_id']) {
+            $is_authenticated = true;
+        }
+        if ($is_authenticated) {
+            $db->query("DELETE FROM likes WHERE comment_id = ? ", array($id));
```

Comment:
Sorry, I did not pushed earlier, but that part was already corrected. Thanks! :bow: